qt/applications/workbench/workbench/plotting/plotscriptgenerator/__init__.py

```diff
@@ -0,0 +1,63 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
+from __future__ import (absolute_import, unicode_literals)
+
+from mantid.plots import MantidAxes
+
+from mantidqt.widgets.plotconfigdialog import curve_in_ax
+from workbench.plugins.editor import DEFAULT_CONTENT
+from workbench.plotting.plotscriptgenerator.axes import generate_add_subplot_command, generate_axis_limit_commands
+from workbench.plotting.plotscriptgenerator.figure import generate_figure_command
+from workbench.plotting.plotscriptgenerator.lines import generate_plot_command
+from workbench.plotting.plotscriptgenerator.utils import generate_workspace_retrieval_commands
+
+FIG_VARIABLE = "fig"
+AXES_VARIABLE = "ax"
+
+
+def generate_script(fig, exclude_headers=False):
+    """
+    Generate a script to recreate a figure.
+
+    This currently only supports recreating artists that were plotted
+    from a Workspace. The format of the outputted script is as follows:
+
+        <Default Workbench script contents (imports)>
```

Comment:
I don't see the default imports in neither the export to clipboard nor to file, it starts from with ADS import and then retrieval. If the Workbench imports were missed it probably answers my other question too

Reply:
Was there a resolution to this that got discussed offline? I too don't see the default content from the editor and my script starts from `from mantid.api import AnalysisDataService as ADS`.

Reply:
I understand now. I get the full imports in the file but the imports for clipboard don't so pasting is easier in workbench without duplicating the headers? I can see arguments for and against this. Let's see how users fair with it. My guess is it will depend on how much they paste inside/outside of workbench.